src/ros2cs/ros2cs_core/WaitSet.cs

```diff
@@ -13,181 +13,465 @@
 // limitations under the License.
```

Comment:
This is quite a substantial change, but I see it would be hard to get as a separate PR.